chalicelib_smaht/checks/helpers/google_utils.py

```diff
@@ -0,0 +1,1107 @@
+            """
+            if increment not in ('daily', 'monthly'):
+                raise IndexError("increment parameter must be one of 'daily', 'monthly'")
+
+            search_results = ff_utils.search_metadata(
+                '/search/?type=TrackingItem&tracking_type=google_analytics&sort=-google_analytics.for_date&limit=1&google_analytics.date_increment=' + increment,
+                key=dict(self.owner.access_key, server=self.owner.server),
+                page_limit=1
+            )
+            if len(search_results) == 0:
+                return None
+
+            iso_date = search_results[0]['google_analytics']['for_date']
+
+            # TODO: Use date.fromisoformat() once we're on Python 3.7
+            year, month, day = iso_date.split('-', 2) # In python, months are indexed from 1 <= month <= 12, not 0 <= month <= 11 like in JS.
+            return date(int(year), int(month), int(day))
+
+
+
+        def fill_with_tracking_items(self, increment):
+            '''
+            This method is meant to be run periodically to fetch/sync Google Analytics data into SMaHT database.
+
+            Adds 1 TrackingItem for each day to represent analytics data for said day.
+            Fill up from latest already-existing TrackingItem until day before current day (to get full day of data).
+
+            TODO:
+            `date.fromisoformat(...)`  is not supported until Python 3.7 though (without extra libraries).
+            See: https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat
+
+            Args:
+                increment - One of 'daily', 'monthly', or 'yearly'. Required.
+            '''
+
+            last_tracking_item_date = self.get_latest_tracking_item_date(increment=increment)
+            tz = pytz.timezone(self.owner.extra_config.get("analytics_timezone", "US/Eastern"))
+            today = datetime.now(tz).date()
+
+            if last_tracking_item_date is None:
+                if increment == 'daily':
+                    # Fill up with last 60 days of Google Analytics data, if no other TrackingItem(s) yet exist.
+                    # TODO: Set this to be ~45 for dev environment & 120 (or more) for production.
+                    last_tracking_item_date = today - timedelta(days=61)
+                    date_to_fill_from = last_tracking_item_date + timedelta(days=1)
+                elif increment == 'monthly':
+                    # Fill up with last 12 months Google Analytics data, if no other TrackingItem(s) yet exist.
+                    month_to_fill_from = today.month - 12
+                    year_to_fill_from = today.year
+                    while month_to_fill_from < 1:
+                        month_to_fill_from = 12 + month_to_fill_from
+                        year_to_fill_from -= 1
+                    date_to_fill_from = date(year_to_fill_from, month_to_fill_from, 1)
+            else:
+                if increment == 'daily':
```

Comment:
Constants could be smart for these in case they change,, applicable in other places as well